### PR TITLE
support <engine_role, write> for disaggregated tiflash

### LINKED ISSUE #6980: support <engine_role, write> for disaggregated tiflash
## Enhancement
We want support `non-disaggregated tiflash` and `disaggregated tiflash` in one cluster.
So for disaggregated-tiflash, compute node should dispatch MPPTask to tiflash whose label contains `<engine, tiflash>` and `<engine_role, write>`.

And for non-disaggregated tiflash, cop task will send to tiflash whose label contains `<engine, tiflash>` but not contain `<engine_role, write>`.

### PR DESCRIPTION
**Pull request by guo-shaoge**
### What problem does this PR solve?

Issue Number: close #6980

Problem Summary: As issue said.

### What is changed and how it works?
Changes mainly in https://github.com/tikv/client-c/pull/128

1. For cop path: change `ResponseIter`'s label_filter
2. For batchCop path: change `StorageDisaggregated.cpp`'s label_filter
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test: endless case will be added later.
- [x] Manual test (add detailed scripts or steps below): will add in endless later.
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: etc/config-template.toml [hunk 1]
```diff
@@ -108,8 +108,12 @@
 ## The minimum duration to handle read-index tasks in each worker.
 # read_index_worker_tick_ms = 10
 # disaggregated_mode = "tiflash_storage" or "tiflash_compute"
-# Means whether we use AutoScaler or PD for tiflash_compute nodes. Default use AutoScaler. Will remove this after AutoScaler is stable.
+## Means whether we use AutoScaler or PD for tiflash_compute nodes. Default use AutoScaler. Will remove this after AutoScaler is stable.
 # use_autoscaler = true
+## True when tiflash_storage has labels like: <engine, tiflash> and <engine_role, write>.
+## False when tiflash_storage only has label: <engine, tiflash>.
+## Only meaningful when disaggregated_mode is 'tiflash_compute'.
+# is_disaggregated_storage_write_role: true
```
Comment (JaySon-Huang) [inline etc/config-template.toml:116]: maybe we don't need this config.

When the compute node setup s3 related items like `storage.s3.bucket`, then send the task to tiflash node with labels `{engine=tiflash, engine_role=write}`.
If not, then send tasks to tiflash node with labels `{engine=tiflash}`

Comment (guo-shaoge) [reply etc/config-template.toml]: Done

**Review by JaySon-Huang — APPROVED** [review]
LGTM

**Review by hehechen — APPROVED** [review]

**Pull request merged**